### ISSUE TITLE
Android Installation: Configuration with name 'default' not found.

### ISSUE THREAD

**Issue by dwilt** [futurice/react-native-audio-toolkit#28]
Getting this error when trying to install the plugin for Android per the [instructions](https://github.com/futurice/react-native-audio-toolkit/blob/master/docs/SETUP.md):

```
FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':app'.
> Cannot evaluate module react-native-audio-toolkit : Configuration with name 'default' not found.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
```

By the way, we should update the docs to tell the user to use `react-native link react-native-audio-toolkit` because it does all of these changes automatically for you for both iOS and Android.

**Comment by dwilt** [comment]
@FruitieX @ohtangza do you guys have some time this week/weekend to do a half hour hangout? I think this plugin could be fantastic, and it's super close but it has some flaws. I'd love to help explain some things that I could seriously improve on the JS side (better importing, promises, etc) as well as talk to you guys about this breaking issue and the other [duration](https://github.com/futurice/react-native-audio-toolkit/issues/27) bug. Thoughts?

**Comment by dwilt** [comment]
I think we could solve a lot of things in a Google Hangout after an hour.

**Comment by lzctbyong** [comment]
me,too

**Comment by lzctbyong** [comment]
solve this issue,you would go to path "android/lib/"
i solved it !

**Comment by dwilt** [comment]
@lzctbyong can you explain more how you solved this? Please provide an
example

**Comment by lzctbyong** [comment]
1.go to "..\node_modules\react-native-audio-toolkit\android" => delete all but  "lib" forlder

2.copy everything in "..\node_modules\react-native-audio-toolkit\android\lib"  to
    "..\node_modules\react-native-audio-toolkit\android"

this way may be can work ,but not a great solve way
@dwilt

**Comment by dwilt** [comment]
Yes, this works. Even easier would be to change the line in `settings.gradle` from:
```
include ':react-native-audio-toolkit'
project(':react-native-audio-toolkit').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-audio-toolkit/android')
```
to
```
include ':react-native-audio-toolkit'
project(':react-native-audio-toolkit').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-audio-toolkit/android/lib')
```

**Comment by dwilt** [comment]
@FruitieX we should update it so that it the `react native link` works properly. Either by moving the files in `lib` to the root of `../node_modules/react-native-audio-toolkit/android/` or by updating the `react-native link` script to pass in the `lib` folder. We also need to update the manual instructions.

**Comment by dwilt** [comment]
Actually, I'm seeing that the manual instructions do call out using the `lib` folder - probably the reason why `react-native link` isn't in the setup instructions.

Since all of the other steps are the same that `react-native link` produces, we should fix this so people could install with `react-native link` as it would reduce install bugs and save people lots of time.

**Comment by ohtangza** [comment]
@dwilt I would love to hear your ideas on how to improve this library. Sadly, I am kind of frantic with launching my app coming next week. So, I am not sure that when I am available for now. Please let me know if @FruitieX and you happen to arrange a Hangout session (put me in a lower priority if I am not necessarily in present). I would try to make myself to participate if I am available. Since I am in Korea, it might be a little trouble to arrange a session because a we all in quite different time zone.

For `Promise` implementation, I also agree on the direction. My app is already in callback hell because we are controlling Player and Recorder alternately. I think we can support both callback and Promise for the time being. It seems to be possible on Android side at least by checking the callback function argument is null or not.

I also agree on the ```lib``` should be moved to ```android``` folder as many other libraries do. I think we could also refer the boilerplate from [react-native-create-library](https://github.com/frostney/react-native-create-library). because the ```react-native new-library``` command seems to be broken according to the open pull request in RN project (https://github.com/facebook/react-native/pull/12235).

**Comment by akademy** [comment]
+1 As a newbie to React Native this bug caused so many headaches...

**Comment by FruitieX** [comment]
Hey, we should definitely update the docs and code so that `react-native link` works. The initial implementation was based on guesswork/trial&error since not much RN documentation was available at the time. If somebody has time to look at this a PR is definitely welcome

**Comment by tmaly1980** [comment]
What exactly is needed to fix `react-native link`? I would like to be able to fix that myself. I have other libraries that depends upon `react-native link` so it's getting called either way, and I made the *wild* assumption that the code put there for this library might actually work. There are issues on the iOS side so I've forked the repo anyway, and I'd like to be able to re-use it without having `react-native link` fail.

**Comment by GeoffreyPlitt** [comment]
This should be mentioned on the README if ```react-native link``` is not fixed.
